velocity/src/main/java/me/leoko/advancedban/velocity/VelocityMethods.java

```diff
@@ -0,0 +1,384 @@
+  }
+
+  @Override
+  public void setupMetrics() {
+    // no bstats for velocity yet
+  }
+
+
+  @Override @Deprecated
+  public boolean isBungee() {
+    return false;
+  }
+
+  @Override
+  public boolean isProxy() {
+    return true;
+  }
+
+  @Override
+  public ServerType getServerType() {
+    return ServerType.VELOCITY;
+  }
+
+  @Override
+  public String clearFormatting(String text) {
+    return text.replaceAll("(&[^\\s])+", "");
+  }
+
+  @Override
+  public Object getPlugin() {
+    return VelocityMain.get();
+  }
+
+  @Override
+  public File getDataFolder() {
+    return dataDirectory.toFile();
+  }
+
+  @Override
+  public void setCommandExecutor(String cmd, TabCompleter tabCompleter) {
+    CommandMeta meta = server.getCommandManager().metaBuilder(cmd).build();
+    server.getCommandManager().register(meta, new CommandReceiverVelocity(server, cmd));
+  }
+
+  @Override
+  public void sendMessage(Object player, String msg) {
+      ((CommandSource) player).sendMessage(LegacyComponentSerializer.legacyAmpersand().deserialize(msg));
+  }
+
+  @Override
+  public String getName(Object player) {
+    if (player instanceof Player) {
+      return ((Player) player).getUsername();
+    } else {
+      return "CONSOLE";
+    }
+  }
+
+  @Override
+  public String getName(String uuid) {
+      return server.getPlayer(UUID.fromString(uuid)).get().getUsername();
+  }
+
+  @Override
+  public String getIP(Object player) {
+    return ((Player)player).getRemoteAddress().getHostName();
+  }
+
+  @Override
+  public String getInternUUID(Object player) {
+    return player instanceof  Player ? ((Player) player).getUniqueId().toString().replace("-", "") : "none";
+  }
+
+  @Override
+  public String getInternUUID(String player) {
+    Optional<Player> optionalPlayer = server.getPlayer(player);
+    return optionalPlayer.map(value -> value.getUniqueId().toString().replace("-", "")).orElse(null);
+  }
+
+  @Override
+  public boolean hasPerms(Object player, String perms) {
+      return ((CommandSource) player).hasPermission(perms);
+  }
+
+  @Override
+  public Permissionable getOfflinePermissionPlayer(String name) {
+    if(luckPermsSupport) return new LuckPermsOfflineUser(name);
+
+    return permission -> false;
+  }
+
+  @Override
+  public boolean isOnline(String name) {
+    return server.getPlayer(name).isPresent();
+  }
+
+  @Override
+  public Player getPlayer(String name) {
+    return server.getPlayer(name).orElse(null);
+  }
+
+  @Override
+  public void kickPlayer(String player, String reason) {
+    server.getPlayer(player).get().disconnect(LegacyComponentSerializer.legacyAmpersand().deserialize(reason));
+  }
+
+  @Override
+  public Player[] getOnlinePlayers() {
+    return server.getAllPlayers().toArray(new Player[]{});
+  }
+
+  @Override
+  public void scheduleAsyncRep(Runnable rn, long l1, long l2) {
+    server.getScheduler().buildTask(getPlugin(), rn).delay(l1*50, TimeUnit.MILLISECONDS).repeat(l2*50, TimeUnit.MILLISECONDS);
```

Comment:
Missing call to .schedule()
```suggestion
    server.getScheduler().buildTask(getPlugin(), rn).delay(l1*50, TimeUnit.MILLISECONDS).repeat(l2*50, TimeUnit.MILLISECONDS).schedule();
```